### ISSUE TITLE
Optimize the rendering of Structure Void/Air with structure blocks

### ISSUE THREAD

**Issue by TelepathicGrunt** [jellysquid3/Sodium#43]
Hello! I was wondering if it would be within the scope of Sodium to optimize the rendering of structure void/air as I have a world with a ton of structure blocks rending those blocks. And well, my performance was at 15 fps when at a render distance of 7, clouds off, fast graphics, yeah. 

I put on lithium, phosphor, and sodium all together and my fps only improved to 20 fps. Even in vanilla, the moment I turn on "show invisible blocks" on structure blocks, my fps just dies lol.

Here's a picture showing the performance dying thanks to Mojang's structure void/air rendering:
![image](https://user-images.githubusercontent.com/40846040/86066549-71e92700-ba40-11ea-8e08-14197397a48b.png)

I hope this can be optimized as it would be immensely helpful!

**Comment by Miros77** [comment]
disable Vsync

**Comment by TelepathicGrunt** [comment]
Disabling V-sync did not solve the issue and I am not sure how it would had anyway.

Pic below is using almost vanilla (no optimization mods) without v-sync:
![image](https://user-images.githubusercontent.com/40846040/87539402-dbd21500-c66b-11ea-9327-2c938fb43a84.png)

Pic with sodium, phosphor, and lithium without v-sync:
![image](https://user-images.githubusercontent.com/40846040/87539744-76caef00-c66c-11ea-80c0-e90fe228dcb1.png)

**Comment by TelepathicGrunt** [comment]
Here's a video demonstrating in a completely blank superflat world in 1.16.2 Fabric with no mods, that a single 48x48x48 structrue block will murder fps when show invisible blocks is turned on
https://streamable.com/qh05b7

Edit: Here's a better video with no slimes and v-sync off.
https://streamable.com/n33vcu

**Comment by Miros77** [comment]
> Here's a video demonstrating in a completely blank superflat world in 1.16.2 Fabric with no mods, that a single 48x48x48 structrue block will murder fps when show invisible blocks is turned on
> https://streamable.com/qh05b7
> 
> Edit: Here's a better video with no slimes and v-sync off.
> https://streamable.com/n33vcu

This should optimize Mojang

**Comment by Titaniumtown** [comment]
Why is it in the pictures where you have low fps, you are using the intergrated graphics, but in those videos, you are using your dedicated graphics card? That may be the issue.

**Comment by TelepathicGrunt** [comment]
The original pictures were before I set minecraft to use the dedicated graphics card but even then, no matter which it uses for graphics, Structure Block's invisible block rendering destroys performance 100% of the time. 

It's not hard to even test it too. Just copy what I did in the video with your rig and it'll struggle with the 48x48x48 size structure block lol

**Comment by TelepathicGrunt** [comment]
I have made an issue report to mojang here as well but it'll take a long time for them to fix it if they ever get around to it. https://bugs.mojang.com/browse/MC-198175

**Comment by TelepathicGrunt** [comment]
I just started doing larger structure pieces and this bug is really killing me lol. It gotten so bad that I couldn't even switch to flying in Creative Mode due to the lag in recognizing key presses. Shame mojang marked this as low priority on their bug tracker 
![image](https://user-images.githubusercontent.com/40846040/94348864-ac496c80-000d-11eb-85ba-87e54071cd6e.png)

**Comment by jellysquid3** [comment]
I've done my best to improve this issue with cb2c9054, but it's not going to be entirely cured by the patch. Ultimately, this geometry needs to be cached so it's not re-built every frame, but there's no mechanism to allow us to do that right now.

Before: ![before.png](https://cdn.discordapp.com/attachments/651120262129123330/772970408869625877/unknown.png)
After: ![after.png](https://cdn.discordapp.com/attachments/651120262129123330/772970426778255390/unknown.png)
